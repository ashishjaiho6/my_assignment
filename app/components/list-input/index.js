import React, {useState} from 'react';
import {StyleSheet, View, TextInput, TouchableOpacity} from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';

export const TaskInputField = props => {
  const [task, setTask] = useState();

  const handleAddTask = value => {
    props.addTask(value);
    setTask(null);
  };

  return (
    <View style={styles.container}>
      <TextInput
        style={styles.inputField}
        value={task}
        onChangeText={text => setTask(text)}
        placeholder={'add an item'}
        placeholderTextColor={'#fff'}
      />
      <TouchableOpacity onPress={() => handleAddTask(task)}>
        <View style={styles.button}>
          <MaterialIcons name="control-point" size={24} color="black" />
        </View>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    borderColor: '#fff',
    backgroundColor: '#3E3364',
    borderWidth: 1,
    marginHorizontal: 20,
    borderRadius: 12,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingHorizontal: 10,
    position: 'absolute',
    bottom: 20,
  },
  inputField: {
    color: '#fff',
    height: 50,
    flex: 1,
  },
  button: {
    height: 30,
    width: 30,
    borderRadius: 5,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
