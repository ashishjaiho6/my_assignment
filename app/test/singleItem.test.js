import React from 'react';
import renderer from 'react-test-renderer';
import {SingleItem} from '../components/single-item';

it(`renders correctly`, () => {
  const tree = renderer.create(<SingleItem task="test" index={1} />);
  expect(tree).toMatchSnapshot();
});
