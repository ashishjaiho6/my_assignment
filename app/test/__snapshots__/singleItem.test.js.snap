// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginHorizontal": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3E3364",
        "borderRadius": 12,
        "height": 50,
        "justifyContent": "center",
        "marginRight": 10,
        "width": 50,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
        }
      }
    >
      1
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3E3364",
        "borderRadius": 12,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 50,
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 16,
          "width": "90%",
        }
      }
    >
      test
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#fff",
              "fontSize": 18,
            },
            Object {
              "marginLeft": 10,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¡²
      </Text>
    </View>
  </View>
</View>
`;
